targets/TARGET_STM/i2c_api.c

```diff
@@ -1728,6 +1754,168 @@ uint32_t i2c_compute_timing(uint32_t clock_src_freq, uint32_t i2c_freq)
 
     return ret;
 }
+#endif /* MBED_CONF_TARGET_I2C_TIMING_VALUE_ALGO */
+
+#ifdef I2C_IP_VERSION_V2
+/**
+ * @brief  Provide the suitable timing depending on requested frequency
+ * @param  i2c Required I2C instance.
+ * @param  current_timing Required I2C timing value.
+ * @param  current_hz Required I2C current hz value.
+ * @param  hz Required I2C bus clock speed.
+ * @retval I2C timing value or 0 in case of error.
+ */
+uint32_t i2c_get_timing(I2CName i2c, uint32_t current_timing, int current_hz,
+                        int hz)
+{
+    uint32_t tim = 0;
+    uint32_t pclk;
+
+    pclk = i2c_get_pclk(i2c);
+
+    if ((current_timing == 0) || (current_hz != hz)) {
+#if defined (I2C_PCLK_DEF)
```

Comment:
While rebasing, would you please :
- replace I2C_PCLK_DEF by I2C_PCLK_XXM and TIMING_VAL_DEFAULT_CLK_100KHZ by TIMING_VAL_64M_CLK_100KHZ
- also store the switch case values in from lowest to highest for enhanced readibility  
Thanks